examples/preprocessing/esg_rm_heart_artefact_pcaobs.py

```diff
@@ -0,0 +1,212 @@
+"""
+.. _ex-pcaobs:
+
+=====================================================================================
+Principal Component Analysis - Optimal Basis Sets (PCA-OBS) removing cardiac artefact
+=====================================================================================
+
+This script shows an example of how to use an adaptation of PCA-OBS
+:footcite:`NiazyEtAl2005`. PCA-OBS was originally designed to remove
+the ballistocardiographic artefact in simultaneous EEG-fMRI. Here, it
+has been adapted to remove the delay between the detected R-peak and the
+ballistocardiographic artefact such that the algorithm can be applied to
+remove the cardiac artefact in EEG (electroencephalogrpahy) and ESG
+(electrospinography) data. We will illustrate how it works by applying the
+algorithm to ESG data, where the effect of removal is most pronounced.
+
+See: https://www.biorxiv.org/content/10.1101/2024.09.05.611423v1
+for more details on the dataset and application for ESG data.
+
+"""
+
+# Authors: Emma Bailey <bailey@cbs.mpg.de>,
+#          Steinn Hauser Magnusson <hausersteinn@gmail.com>
+# License: BSD-3-Clause
+# Copyright the MNE-Python contributors.
+
+import glob
+
+import numpy as np
+
+###############################################################################
+# Download sample subject data from OpenNeuro if you haven't already
+# This will download simultaneous EEG and ESG data from a single participant after
+# median nerve stimulation of the left wrist
+# Set the target directory to your desired location
+import openneuro as on
+from matplotlib import pyplot as plt
+
+import mne
+from mne import Epochs, concatenate_raws, events_from_annotations
+from mne.io import read_raw_eeglab
+from mne.preprocessing import find_ecg_events, fix_stim_artifact
+
+# add the path where you want the OpenNeuro data downloaded. Files total around 8 GB
+# target_dir = "/home/steinnhm/personal/mne-data"
+target_dir = "/data/pt_02569/test_data"
```

Comment:
I know it's not an ideal dataset perhaps, but could you show the algorithm on `sample` instead? Then it could just be another subsection in

https://github.com/mne-tools/mne-python/blob/main/tutorials/preprocessing/50_artifact_correction_ssp.py

Although this algorithm isn't SSP, I think it's related enough to put at the bottom of that tutorial as a "hey you can also use this SSP-like algorithm ... see that it works similarly for this dataset ... but in general is better for datasets with few electrodes".

I think it would also be more discoverable there since I think more people will look at that tutorial when asking themselves the question "how do I get rid of (heart) artifacts"